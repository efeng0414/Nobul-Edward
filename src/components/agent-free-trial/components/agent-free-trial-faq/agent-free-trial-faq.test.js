import React from "react";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";
import AgentFreeTrialFaqWrapped from "./index";
import {
  freeTrialFaq,
  renderFreeTrialFaqItem
} from "../../../../utilities/free-trial";

describe("AgentFreeTrialFaq Component", () => {
  const { WrappedComponent: AgentFreeTrialFaq } = AgentFreeTrialFaqWrapped;
  const componentProps = {
    intl: global.getIntl()
  };

  const componentJSX = <AgentFreeTrialFaq {...componentProps} />;
  describe("Component Snapshot", () => {
    const snapshotJSX = (
      <IntlProvider {...global.intl}>{componentJSX}</IntlProvider>
    );
    const componentSnapshot = renderer.create(snapshotJSX).toJSON();
    it("Should match stored snapshot", () => {
      expect(componentSnapshot).toMatchSnapshot();
    });
  });

  describe("Component Render", () => {
    const componentMount = global.Enzyme.shallow(componentJSX);
    const Obj = freeTrialFaq({ intl: componentProps.intl });
    const expectedObject = Obj.map(renderFreeTrialFaqItem);
    it("should have  correct classname in the rendered jsx", () => {
      expect(
        componentMount
          .find("div")
          .at(3)
          .hasClass(expectedObject[0].props.className)
      ).toEqual(true);
    });
  });
});
