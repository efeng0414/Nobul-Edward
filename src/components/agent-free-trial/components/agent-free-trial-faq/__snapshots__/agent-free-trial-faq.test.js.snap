// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentFreeTrialFaq Component Component Snapshot Should match stored snapshot 1`] = `
<div
  className="free-trial-faq"
>
  <div
    className="free-trial-faq-title"
  >
    Frequently Asked Questions
  </div>
  <div
    className="free-trial-faq-container"
  >
    <div
      className="free-trial-faq-item"
    >
      <div
        className="free-trial-faq-item-title"
      >
        Will I be charged during my free trial?
      </div>
      <div
        className="free-trial-faq-item-body"
      >
        You will not be charged any upfront fees. Once your free trial begins, you will have 90 days to sample all that nobul premium has to offer â€” for FREE.
      </div>
    </div>
    <div
      className="free-trial-faq-item"
    >
      <div
        className="free-trial-faq-item-title"
      >
        What happens at the end of my free trial?
      </div>
      <div
        className="free-trial-faq-item-body"
      >
        Your membership will automatically renew at the end of your free trial. You can cancel at any time prior to the renewal at nobul.com/my-dashboard/settings.
      </div>
    </div>
    <div
      className="free-trial-faq-item"
    >
      <div
        className="free-trial-faq-item-title"
      >
        Can I change or cancel my plan later on?
      </div>
      <div
        className="free-trial-faq-item-body"
      >
        Yes. You can downgrade or cancel anytime from your settings page. When you cancel, you will lose your Premium features at the end of your billing cycle.
      </div>
    </div>
  </div>
</div>
`;
